x-pack/plugins/security/server/lib/authentication/__tests__/authenticator.js

```diff
@@ -102,7 +102,7 @@ describe('Authenticator', () => {
       }
     });
 
-    it('fails if all authentication providers fail.', async () => {
+    it('fails if any authentication providers fail.', async () => {
```

Comment:
Unfortunately we can't really test the distinction between `any` and `all` without fundamental changes to how providers are configured for the authenticator. Since they are currently hardcoded as an implementation detail that cannot be modified at runtime, there's no way to configure mock providers that verify the nuances of failure handling in different combinations of providers.

When we make auth providers pluggable, this problem will go away and we'll be able to test this properly.